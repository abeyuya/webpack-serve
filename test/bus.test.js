const weblog = require('webpack-log');

const eventbus = require('../lib/bus');

describe('bus', () => {
  weblog({ id: 'webpack-serve', level: 'silent', name: 'serve' });

  test('should subscribe to events in options', async () =>
    new Promise((resolve) => {
      const bus = eventbus({
        on: {
          foo: () => {
            expect(true);
            resolve();
          },
        },
      });

      bus.emit('foo');
    }));

  test('should not allow non-object options', () => {
    const fn = () => {
      eventbus({ on: 'foo' });
    };
    expect(fn).toThrowErrorMatchingSnapshot();
  });

  test('should not allow a non-function handler', () => {
    const fn = () => {
      eventbus({
        on: {
          foo: 'bar',
        },
      });
    };
    expect(fn).toThrowErrorMatchingSnapshot();
  });
});
