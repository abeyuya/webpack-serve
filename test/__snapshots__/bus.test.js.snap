// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bus should not allow a non-function handler 1`] = `"The value for an \`on\` event handler must be a Function. event: foo"`;

exports[`bus should not allow non-object options 1`] = `"The value for the \`on\` option must be an Object. Please see the README."`;
